pkg/dhctl/apply.go

```diff
@@ -2,71 +2,85 @@ package dhctl
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
 	"github.com/spf13/cobra"
 	"github.com/tislib/data-handler/pkg/dhctl/flags"
 	"github.com/tislib/data-handler/pkg/formats/batch"
 	"github.com/tislib/data-handler/pkg/formats/hclformat"
-	yamlformat "github.com/tislib/data-handler/pkg/formats/yaml"
-	"log"
-	"os"
-	"strings"
+	"github.com/tislib/data-handler/pkg/formats/yaml"
 )
 
 var applyCmd = &cobra.Command{
 	Use:   "apply",
 	Short: "apply - apply resources",
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
 		parseRootFlags(cmd)
 
-		file, err := cmd.Flags().GetString("file")
-		check(err)
+		inputFilePath, err := cmd.Flags().GetString("file")
+		if err != nil {
+			return fmt.Errorf("failed to get input file path: %w", err)
+		}
+		if inputFilePath == "" {
+			return errors.New("file must be provided")
+		}
 
-		migrate, err := cmd.Flags().GetBool("migrate")
-		check(err)
+		doMigration, err := cmd.Flags().GetBool("migrate")
+		if err != nil {
+			return fmt.Errorf("failed to get migration flag: %w", err)
+		}
 
 		dataOnly, err := cmd.Flags().GetBool("data-only")
-		check(err)
+		if err != nil {
+			return fmt.Errorf("failed to get data-only flag: %w", err)
+		}
 
 		force, err := cmd.Flags().GetBool("force")
-		check(err)
-
-		var overrideConfig = new(flags.OverrideConfig)
-		overrideFlags.Parse(overrideConfig, cmd, args)
-
-		if file == "" {
-			log.Fatal("file should provided")
+		if err != nil {
+			return fmt.Errorf("failed to get force flag: %w", err)
+			return err
 		}
 
-		if strings.HasSuffix(file, ".hcl") {
-			in, err := os.Open(file)
+		overrideConfig := new(flags.OverrideConfig)
+		if err := overrideFlags.Parse(overrideConfig, cmd, args); err != nil {
```

Comment:
`overrideFlags.Parse(overrideConfig, cmd, args) ` - is not returning error, so please call it without if checking. 
just `overrideFlags.Parse(overrideConfig, cmd, args)`